### PR TITLE
fix(did-provider-pkh): refactor and simplify did:pkh plugin

### PR DESCRIPTION
**Pull request by dchagastelles**
## What issue is this PR fixing
There's no issue being fixed, just small improvements on did-provider-pkh plugin:

- minor parameters corrections
- use of namespace parameter
- allow passing privateKey to import key directly on identifier creation

## What is being changed
Only changing did-provider-pkh plugin. 

## Quality
Check all that apply:
* [X] I want these changes to be integrated
* [X] I successfully ran `pnpm i`, `pnpm build`, `pnpm test`, `pnpm test:browser` locally.
* [X] I allow my PR to be updated by the reviewers (to speed up the review process).
* [ ] I added unit tests. 
* [X] I added integration tests. [were already there]
* [ ] I did not add automated tests because _________, and I am aware that a PR without tests will likely get rejected.

### CODE REVIEW

File: packages/did-provider-pkh/src/pkh-did-provider.ts [hunk 1]
```diff
@@ -1,135 +1,164 @@
...
+export interface CreateDidPkhOptions {
+  namespace: string;
+  privateKey: string;
   /**
    * This can be hex encoded chain ID (string) or a chainId number
    *
    * If this is not specified, `1` is assumed.
    */
-  chainId?: string | number
+  chainId?: string;
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/pkh-did-provider.ts:34]: Why not leave this as a number as well?
```suggestion
  chainId?: string | number;
```

File: packages/did-provider-pkh/src/resolver.ts [hunk 2]
```diff
@@ -32,29 +32,21 @@ function toDidDoc(did: string, accountId: string): any {
...
     ],
     authentication: [vmId],
     assertionMethod: [vmId],
+  };
+  if (!isValidNamespace(namespace)) {
+    console.error(
+      `Invalid namespace '${namespace}'. Valid namespaces are: ${SECPK1_NAMESPACES}`
+    );
+    throw new Error(
+      `Invalid namespace '${namespace}'. Valid namespaces are: ${SECPK1_NAMESPACES}`
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/resolver.ts:46]:
```suggestion
      `illegal_argument: namespace '${namespace}' not supported. Valid namespaces are: ${SECPK1_NAMESPACES}`
```

File: packages/did-provider-pkh/src/pkh-did-provider.ts [hunk 3]
```diff
@@ -1,135 +1,164 @@
...
-type IContext = IAgentContext<IKeyManager>
+const isIn = <T>(values: readonly T[], value: any): value is T => {
+  return values.includes(value);
+};
+
+export const SECPK1_NAMESPACES = ['eip155'] as const;
+export const isValidNamespace = (x: string) => isIn(SECPK1_NAMESPACES, x);
 
 /**
  * Options for creating a did:ethr
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/pkh-did-provider.ts:23]:
```suggestion
 * Options for creating a did:pkh
```

**Review by mirceanis — CHANGES_REQUESTED** [review]
Thanks for contributing this.

Since this is meant to clean up some code, I also made some notes that should make this align better with the patterns we use in other parts of the codebase.
It's mostly about error messages and debug output. Please take a look.

File: packages/did-provider-pkh/src/pkh-did-provider.ts [hunk 4]
```diff
@@ -1,135 +1,164 @@
...
   ): Promise<Omit<IIdentifier, 'provider'>> {
+    const namespace = options?.namespace ? options.namespace : 'eip155';
 
-    const key = await context.agent.keyManagerCreate({ kms: kms || this.defaultKms, type: 'Secp256k1' })
-    const publicAddress = toEthereumAddress(key.publicKeyHex);
-
-    const network = options?.chainId;
-    if (!network) {
+    if (!isValidNamespace(namespace)) {
+      console.error(
```
[the remark this answers is not in the document]
Comment (dchagastelles) [reply packages/did-provider-pkh/src/pkh-did-provider.ts]: done

File: packages/did-provider-pkh/src/pkh-did-provider.ts [hunk 5]
```diff
@@ -1,135 +1,169 @@
...
+  ManagedKeyInfo,
+} from '@veramo/core';
+
+import { AbstractIdentifierProvider } from '@veramo/did-manager';
+import Debug from 'debug'
 
-import { AbstractIdentifierProvider } from '@veramo/did-manager'
-import { computePublicKey } from '@ethersproject/signing-key'
-import { BigNumber } from '@ethersproject/bignumber'
+const debug = Debug('veramo:data-store:migrate-presentation-issuance-date')
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/pkh-did-provider.ts:14]:
```suggestion
const debug = Debug('veramo:pkh-did-provider')
```

File: packages/did-provider-pkh/src/pkh-did-provider.ts [hunk 6]
```diff
@@ -1,135 +1,169 @@
...
+      const identifier: Omit<IIdentifier, 'provider'> = {
+        did: 'did:pkh:' + namespace + ':' + this.chainId + ':' + evmAddress,
+        controllerKeyId: key.kid,
+        keys: [key],
+        services: [],
+      };
+      return identifier;
+    } else {
+      debug('Could not create identifier due to some errors');
+      throw new Error('Could not create identifier due to some errors');
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/pkh-did-provider.ts:106]:
```suggestion
      throw new Error('unknown_error: could not create identifier due to errors creating or importing keys');
```

File: packages/did-provider-pkh/src/resolver.ts [hunk 7]
```diff
@@ -1,29 +1,27 @@
...
-} from 'did-resolver'
-
-const DID_LD_JSON = 'application/did+ld+json'
-const DID_JSON = 'application/did+json'
-const SECPK1_NAMESPACES = ['eip155', 'bip122']
-const TZ_NAMESPACE = 'tezos'
+  Resolvable,
+  ResolverRegistry,
+} from 'did-resolver';
+import { isValidNamespace, SECPK1_NAMESPACES } from './pkh-did-provider';
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/resolver.ts:10]:
```suggestion
import Debug from 'debug'

const debug = Debug('veramo:pkh-did-resolver')
```

File: packages/did-provider-pkh/src/resolver.ts [hunk 8]
```diff
@@ -32,29 +30,21 @@ function toDidDoc(did: string, accountId: string): any {
...
         controller: did,
-        blockchainAccountId: accountId,
+        blockchainAccountId,
       },
     ],
     authentication: [vmId],
     assertionMethod: [vmId],
+  };
+  if (!isValidNamespace(namespace)) {
+    console.error(
```
Comment (mirceanis) [inline packages/did-provider-pkh/src/resolver.ts:40]:
```suggestion
    debug(
```

**Review by mirceanis — CHANGES_REQUESTED** [review]
Thanks for the updates.
I suggested some more tiny fixes.

**Review by mirceanis — APPROVED** [review]

**Pull request merged**